src/index.spec.ts

```diff
@@ -1,12 +1,196 @@
+  test.each(productNameOverrides)(
+    'throws an error with invalid product name %p',
+    (name) => {
+      const product = createTestProduct({ name });
+      expect(() => createProduct(product)).toThrow();
+    },
+  );
+
+  const productDescOverrides = [[''], [faker.random.alphaNumeric(31)]];
+  test.each(productDescOverrides)(
+    'throws an error with invalid product description %p',
+    (description) => {
+      const product = createTestProduct({ description });
+      expect(() => createProduct(product)).toThrow();
+    },
+  );
+
+  const productPriceOverrides = [[-10], [-1.453]];
+  test.each(productPriceOverrides)(
+    'throws an error with with invalid product price %p',
+    (price) => {
+      const product = createTestProduct({ price });
+      expect(() => createProduct(product)).toThrow();
+    },
+  );
+
+  const productTagsOverrides = [[[]], [['tag1', 'tag2']]];
+  test.each(productTagsOverrides)(
+    'throws an error with invalid product tags %p',
+    (tags: string[]) => {
+      const product = createTestProduct({ tags });
+      expect(() => createProduct(product)).toThrow();
+    },
+  );
+});
+
+describe('create random product', () => {
+  test('fails if no user exists with the provided email', () => {
+    const email = 'unperson@users.com';
+    expect(() => createRandomProduct(email)).toThrowErrorMatchingInlineSnapshot(
+      `"You are not allowed to create products"`,
+    );
+  });
+
+  test("fails if user doesn't have the creator role", () => {
+    const email = users.find((it) => it.role !== 'creator')!!.email;
+    expect(() => createRandomProduct(email)).toThrowErrorMatchingInlineSnapshot(
+      `"You are not allowed to create products"`,
+    );
+  });
+
+  test('returns a new product when user is a creator', () => {
+    const email = users.find((it) => it.role === 'creator')!!.email;
+    const product = createRandomProduct(email);
+
+    expect(product.id).toBeDefined();
+    expect(product.name).toBeDefined();
+    expect(product.description).toBeDefined();
+    expect(product.tags).toBeDefined();
+    expect(product.price).toBeDefined();
+  });
+});
+
+describe('getting Star Wars planets', () => {
+  const mockedFetch = fetch as jest.MockedFunction<typeof fetch>;
+
+  test('works when response successfully returns a JSON', async () => {
+    const jsonBody = {};
+    const body = JSON.stringify(jsonBody);
+    const response = new Response(body, { status: 200 });
+    mockedFetch.mockResolvedValueOnce(response);
+
+    const receivedJson = await getStarWarsPlanets();
+
+    expect(receivedJson).toEqual(jsonBody);
+    expect(mockedFetch).toHaveBeenCalledTimes(1);
+    expect(mockedFetch).toHaveBeenCalledWith('https://swapi.dev/api/planets');
+  });
+
+  test('throws an error when response fails for HTTP error', async () => {
+    const response = new Response('', { status: 400 });
+    mockedFetch.mockResolvedValueOnce(response);
```

Comment:
In this case, you want to use `.mockRejectedValueOnce`, this test passes, but it is always related to not being able to parse the json